agent/hcp/manager.go

```diff
@@ -71,15 +77,80 @@ func NewManager(cfg ManagerConfig) *Manager {
 	}
 }
 
+// runReporter initializes, and if successful, then runs the metrics reporter.
+func (m *Manager) runReporter(ctx context.Context) error {
+	// Step 1: Obtain CCM telemetry configuration
+	// Only enable HCP metrics reporting if server is registered with management plane.
+	telemetryCfg, err := m.cfg.Client.FetchTelemetryConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to obtain CCM telemetry config: %v", err)
+	}
+
+	if telemetryCfg == nil {
```

Comment:
Thought I should handle the case where we don't have an error per say, but the server isn't registered with CCM 🤔 This might not be nil, more empty (no endpoint) but for now since the CCM protos are not yet available I did it this way. Might change later.